android/src/main/java/com/swmansion/reanimated/NodesManager.java

```diff
@@ -445,7 +448,13 @@ public void onEventDispatch(Event event) {
       handleEvent(event);
       performOperations();
     } else {
-      mEventQueue.offer(event);
+      boolean shouldSaveEvent = false;
+      String eventName = mCustomEventNamesResolver.resolveCustomEventName(event.getEventName());
+      int viewTag = event.getViewTag();
+      String key = viewTag + eventName;
+
+      shouldSaveEvent |= (mCustomEventHandler != null && mNativeProxy.isAnyHandlerWaitingForEvent(key));
+      mEventQueue.offer(new CopiedEvent(event));
```

Comment:
Copied from #1149 
> Maybe we could also pool CopiedEvent objects same way we have pools of other events. In this case this should be rather efficient as there typically won't be more than 2-3 events in use.

Reply:
I don't see a need for a CopiedEvent pool.

Reply:
Not using the pool will strain GC, event objects will be created and removed frequently if events will be coming from other source than the UI thread.

Reply:
It doesn't make sense for me because we still have to create and remove events' payloads.

Reply:
So maybe @kmagiera will chime in since he suggested this change in the original PR.